### ISSUE TITLE
function for re-referencing

### ISSUE THREAD

**Issue by gavinmischler** [naplab/naplib-python#120]
## Expected Behavior
General function which ideally should take in a graph (in matrix form), defining what electrodes to use when computing reference for that electrode. It should also take a method, such as 'avg', 'med', or 'pca'

For example, if 'resp' has 4 electrodes,
```python
rereferenced_data = naplib.preprocessing.rereference(data, field='resp', arr=arr, method='avg')
```
where ``arr`` is a 4x4 matrix:

to define blocks for local rereferencing:
```python
arr = [[1,1,0,0],
       [1,1,0,0],
       [0,0,1,1],
       [0,0,1,1]]
```
to define blocks for global rereferencing:
```python
arr = [[1,1,1,1],
       [1,1,1,1],
       [1,1,1,1],
       [1,1,1,1]]
```
to define blocks for a weighted rereferencing:
```python
arr = [[1,.5,.25,0],
       [.5,1,.5,.25],
       [.25,.5,.5,.5],
       [0,.25,.5,1]]
```

### This would likely require another set of functions which help a user define these matrices based on simpler things.

For example

```python
arr = naplib.utils.create_block_rereference_arr([1,1,2,2])
# arr = [[1,1,0,0],
#        [1,1,0,0],
#        [0,0,1,1],
#        [0,0,1,1]]

arr = naplib.utils.create_block_rereference_arr([1,1,1,1])
# arr = [[1,1,1,1],
#        [1,1,1,1],
#        [1,1,1,1],
#        [1,1,1,1]]
```

or to get it from a list/array of channel names:

```python
arr = naplib.utils.create_rereference_arr_from_channelnames(['RTx1', 'RTx2', 'RTs1', 'RTs2'], method='amplifier')
# arr = [[1,1,0,0],
#        [1,1,0,0],
#        [0,0,1,1],
#        [0,0,1,1]]

```

**Comment by menoua** [comment]
Why is this not closed automatically? Is there something left for this?

**Comment by menoua** [comment]
This was fixed by #122. Closing it.
